crates/scarb-artifacts/src/lib.rs

```diff
@@ -0,0 +1,558 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs;
+use test_collector::LinkedLibrary;
+
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetArtifacts {
+    version: u32,
+    contracts: Vec<StarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetContract {
+    id: String,
+    package_name: String,
+    contract_name: String,
+    artifacts: StarknetContractArtifactPaths,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+struct StarknetContractArtifactPaths {
+    sierra: Utf8PathBuf,
+    casm: Utf8PathBuf,
+}
+
+/// Contains compiled Starknet artifacts
+#[derive(Debug, PartialEq, Clone)]
+pub struct StarknetContractArtifacts {
+    /// Compiled sierra code
+    pub sierra: String,
+    /// Compiled casm code
+    pub casm: String,
+}
+
+/// Get deserialized contents of `starknet_artifacts.json` file generated by Scarb
+///
+/// # Arguments
+///
+/// * `path` - A path to `starknet_artifacts.json` file.
+fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
+    let starknet_artifacts =
+        fs::read_to_string(path).with_context(|| format!("Failed to read {path:?} contents"))?;
+    let starknet_artifacts: StarknetArtifacts =
+        serde_json::from_str(starknet_artifacts.as_str())
+            .with_context(|| format!("Failed to parse {path:?} contents. Make sure you have enabled sierra and casm code generation in Scarb.toml"))?;
+    Ok(starknet_artifacts)
+}
+
+/// Try getting path to Scarb starknet artifacts for the given package
+///
+/// Try getting the path to `starknet_artifacts.json` file that is generated by `scarb build` command.
+/// If the file is not present, for example when package doesn't define the starknet target, `None` is returned.
+///
+/// # Arguments
+///
+/// * `path` - A path to the Scarb package
+/// * `target_name` - A name of the target that is being built by Scarb
+pub fn try_get_starknet_artifacts_path(
+    target_dir: &Utf8Path,
+    target_name: &str,
+) -> Result<Option<Utf8PathBuf>> {
+    let path = target_dir.join("dev");
+    let paths = fs::read_dir(path);
+    let Ok(mut paths) = paths else {
+        return Ok(None);
+    };
+    let starknet_artifacts = paths.find_map(|path| match path {
+        Ok(path) => {
+            let name = path.file_name().into_string().ok()?;
+            (name == format!("{target_name}.starknet_artifacts.json")).then_some(path.path())
+        }
+        Err(_) => None,
+    });
+    let starknet_artifacts: Option<Result<Utf8PathBuf>> = starknet_artifacts.map(|artifacts| {
+        Utf8PathBuf::try_from(artifacts.clone())
+            .with_context(|| format!("Failed to convert path = {artifacts:?} to Utf8PathBuf"))
+    });
+    starknet_artifacts.transpose()
+}
+
+/// Get the map with `StarknetContractArtifacts` for the given package
+///
+/// # Arguments
+///
+/// * `path` - A path to the Scarb package
```

Comment:
It is a path to `{target_name}.starknet_artifacts.json` file, idk why it was described like this before 🤔